### DISCUSSION TITLE
Disable default namespace

### DISCUSSION THREAD

**Discussion by rChaoz** [socketio/socket.io#4973] [Q&A] [answered]
I'm currently trying to migrate from a custom websocket implementation to socket.io.
I have multiple endpoints (that are never used in parallel), which I want to transform into namespaces.

My question is very simple: since I don't need it, what's the best way to completely disable the default namespace? Middleware that always calls `next(error)`? Calling `disconnect(true)` in `on('connection')`? I would like to know if there is a "standard" way of doing this.

**Comment by darrachequesne** [comment, marked as answer]
Hi! Yes, I think always calling `next(error)` is the best option:

```js
io.use((_, next) => next(new Error("unauthorized")));
```

There is currently no way to properly delete a namespace.
